### PR TITLE
Add intersections

### PR DESCRIPTION
**Pull request by luxx**
Initial javascript implementation of path intersections. It takes two time-sorted lists (Path objects) and computes the intersections. Supports padding for time and location. Since the lists are sorted, it just is a matter of maintaining two pointers and walking through both lists. I put in a couple test cases, but it is possible I missed some edge cases. 

Some items we can easily adjust as the rest of the project moves along:
- for input I am using epoch time, need to figure out what the server will be sending
- intersections return time on the first path + distance, however these might need to be batched or throttled in some way for user display, not sure yet about the requirements.

### CODE REVIEW

File: CovidGuardianClient/lib/PathHelpers.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+
+// return true if a = b ± padding
+export function fuzzyEqual(a, b, padding) {
+  return (a <= b && b - Math.abs(padding) <= a) || (a > b && b + Math.abs(padding) >= a);
```
Comment (BranescuMihai) [inline CovidGuardianClient/lib/PathHelpers.js]: return Math.abs(a-b) <= Math.abs(padding) to simplify a bit?

File: CovidGuardianClient/lib/__tests__/PathHelpers-test.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  let seattle = { lat: 47.609755, long: -122.337793 }
+  let helsinki = { lat: 60.169840, long: 24.938330 }
+
+  it(`returns 0 for same location`, () => {
+    expect(distance(seattle.lat, seattle.long, seattle.lat, seattle.long)).toEqual(0);
+  });
+  
+  it(`returns for distant locations`, () => {
+    expect(distance(seattle.lat, seattle.long, helsinki.lat, helsinki.long)).toEqual(7670.988740099061);
+  });
```
Comment (BranescuMihai) [inline CovidGuardianClient/lib/__tests__/PathHelpers-test.js]: Maybe add a test for a really close example, like 10-20 meters, since that's our primary use case

File: CovidGuardianClient/lib/PathHelpers.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+// return distance in km between lat/long points
+// credit to https://stackoverflow.com/questions/27928/calculate-distance-between-two-latitude-longitude-points-haversine-formula
+export function distance(lat1, lon1, lat2, lon2) {
```
Comment (BranescuMihai) [inline CovidGuardianClient/lib/PathHelpers.js]: Checked this with a few examples plus difference from the method used by [google maps api ](https://cloud.google.com/blog/products/maps-platform/how-calculate-distances-map-maps-javascript-api) (below 0.01 meters difference for around 1km distance), so seems perfectly fine!

**Review by BranescuMihai — APPROVED** [review]
Looks good to me!

File: CovidGuardianClient/lib/PathHelpers.js (hunk 1 above)
Comment (luxx) [reply CovidGuardianClient/lib/PathHelpers.js]: fixed

File: CovidGuardianClient/lib/__tests__/PathHelpers-test.js (hunk 2 above)
Comment (luxx) [reply CovidGuardianClient/lib/__tests__/PathHelpers-test.js]: fixed
